general/development/tools/behat/writing.md

```diff
@@ -0,0 +1,401 @@
+
+:::
+
+Where you have several Given, When, and Then steps you should use the words **And**, and **But**, for example:
+
+```gherkin
+Given the following user exists:
+  | username   | ccolon             |
+  | First name | Colin              |
+  | Last name  | Colon              |
+  | email      | ccolon@example.com |
+And the following course exists:
+  | Name      | Jump Judging (Level 1) |
+  | Shortname | sjea1                  |
+When I log in as "ccolon"
+And I navigate to "Site home > Jump Judging (Level 1)"
+Then I should see "You are not enrolled in this course"
+But I should see "Enrol now"
+```
+
+To initialize and run your tests, please follow the instructions of [Running acceptance test](/general/development/tools/behat/running).
+
+## Create your own tests
+
+Behat tests are located within the directory tests/behat of your plugin.
+The different tests are defined in files with the ending *.feature.
+First, you have to define the header of your test:
+
+```gherkin
+@mod @mod_yourplugin @javascript
+Feature: Here comes a description of your user story.
+```
+
+The tags on top of the feature description can be used to select specific test cases when running the tests.
+The '@JavaScript' tag should only be used, if JavaScript is needed to execute your test. This is dependent on the step you will use in your definition.
+JavaScript tests are usually much slower than tests executed without JavaScript.
+
+Afterwards you can specify a scenario:
+
+```gherkin
+@javascript
+Scenario: Description of your scenario, which you want to test.
+    When I log in as "student1"
+    And I am on "Course 1" course homepage
+```
+
+Again you can define specific tags. Afterwards you write the steps, which should be executed during your test.
+
+### Multiple Scenarios
+
+You can have an arbitrary amount of scenarios within a test. Please make sure they all belong to the same feature.
+If you have certain steps, which should be executed for every scenario of a feature, you can define them using a background:
+
+```gherkin
+Background:
+    Given the following "courses" exist:
+      | fullname | shortname | category | groupmode |
+      | Course 1 | C1        | 0        | 1         |
+    And the following "users" exist:
+      | username | firstname | lastname | email |
+      | teacher1 | Theo | Teacher | teacher1@example.com |
+```
+
+This is usually used, to define the different **GIVEN** steps.
+
+### Use existing steps
+
+There are different ways how to effectively browse the available existing steps:
+
+#### Moodle Administration
+
+Moodle offers within its administration menu under Site Administration > Development > Acceptance Testing a complete and searchable list of all available step definitions.
+However, make sure you installed the behat test site first!
+
+#### PhpStorm
```

Comment:
I think it would be better to create a generic reference to IDE's (to avoid specific tools). That way, it will be easier to maintain ;-)